### ISSUE TITLE
Problem adding Sofar inverter to Homey

### ISSUE THREAD

**Issue by timsonder** [AdyRock/ady.solarman#16]
Hi Adrian,
I see similar messages about this but anyway :) I'm having trouble adding solarpanels with an Sofar Inverter to Homey.
Error message:
<img width="428" alt="image" src="https://github.com/AdyRock/ady.solarman/assets/7859948/8063d922-b7b9-4212-a25b-58136f51670f">
The Log says:
* 2024-01-11T07:48:27.263Z Found Inverter: : 192.168.50.205, NaN * 2024-01-11T07:48:27.266Z Checking register 14 for grid frequency: * 2024-01-11T07:48:32.284Z Returned null. Checking register 1156 for grid frequency: * 2024-01-11T07:48:37.315Z Returned null. Checking register 524 for grid frequency: * 2024-01-11T07:48:42.337Z Returned null. No suitable inverters found 
The inverter is this one:
![787F8BA0-58CF-47C4-B225-22372EC4DD73_1_105_c](https://github.com/AdyRock/ady.solarman/assets/7859948/a7b06add-136c-4d40-86df-2b3cfa1a7946)
![IMG_0228](https://github.com/AdyRock/ady.solarman/assets/7859948/cd5de82a-fc6f-44d5-8720-0edf19c7b0d7)

Hope you're able to help :) Many thanks in advance,
Tim

**Comment by horatiuanghel** [comment]
Hi
I have the same issue. Did you found any solution?

**Comment by timsonder** [comment]
I did not get a response yet so no unfortunately not.

**Comment by AdyRock** [comment]
I haven't managed to track down a register map for the 2700TL inverter.
If you want to try and find the registers, you can use the Send Command tab in the app settings page. Enter a number between 1 and 65535 in the register field and tap on Read Reg. The result is shown in the Response panel.
The result will always be an integer, so a factor will need to be applied. For example, the grid frequency will likely come back as 5000, which represents 50.00.

**Comment by timsonder** [comment]
Hi Adrian,
I tried that but I get the same response no matter what number I put in, like:
<img width="482" alt="image" src="https://github.com/AdyRock/ady.solarman/assets/7859948/aee3bc33-97cf-400e-859f-c559a69f8a20">
Any other suggestions?

**Comment by Stable-Adrian** [comment]
Could you Enable the Diagnostic log, try one of the registers and then send the log.

**Comment by timsonder** [comment]
Hi Adrian, I sent the log although I've put this URL in the address. Please let me know if the log finds its way to you or let me know what e-mail address I could send it to. Thank you!

**Comment by AdyRock** [comment]
I have the log. Did you try getting a register as I only see the automated detection fetches?

**Comment by timsonder** [comment]
Yes I tried however I'm not sure how to send you this log but the response is always the same, no matter what number I put in:
<img width="462" alt="image" src="https://github.com/AdyRock/ady.solarman/assets/7859948/2459cae1-703b-47fd-a6f9-78c011afe1c6">
